[turborepo] ESLint TypeError: Cannot convert undefined or null to object with eslint-plugin-turbo 1.10.4

### What version of Turborepo are you using?

1.8.3

### What package manager are you using / does the bug impact?

npm

### What operating system are you using?

Mac

### Describe the Bug

Attempting to run ESLint with eslint-plugin-turbo@1.10.4 throws errors and exits anywhere I'm using `process.env`. This just started happening, and appears to be related to the latest release of the plugin because the previous version we were using (1.9.8) is working as expected.

```
Oops! Something went wrong! :(

ESLint: 7.32.0

TypeError: Cannot convert undefined or null to object
Occurred while linting <root>/packages/design-system/api/getTranscript.ts:3
    at Function.values (<anonymous>)
    at xe.test (<root>/node_modules/eslint-plugin-turbo/dist/index.js:84:5623)
    at a (<root>/node_modules/eslint-plugin-turbo/dist/index.js:84:6743)
    at MemberExpression (<root>/node_modules/eslint-plugin-turbo/dist/index.js:84:7630)
    at <root>/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (<root>/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (<root>/node_modules/eslint/lib/linter/node-event-generator.js:293:26)
    at NodeEventGenerator.applySelectors (<root>/node_modules/eslint/lib/linter/node-event-generator.js:322:22)
    at NodeEventGenerator.enterNode (<root>/node_modules/eslint/lib/linter/node-event-generator.js:336:14)
```

### Expected Behavior

It should not exit

### To Reproduce

Install ESLint (we're using 7.32.0, haven't tried updating yet) and eslint-plugin-turbo@1.10.4. Then configure ESLint to use the turbo plugin and run it against any file containing `process.env` (example: `const remoteVars = (process.env.REMOTES || {}) as Record<string, () => Promise<any>>`)

I'll see what happens if we bump the ESLint version in the meantime...

#### UPDATE (T+5 minutes)

Upgraded to ESLint 8.43.0 and I'm getting a better error now but it's still failing in the same way:

```
Oops! Something went wrong! :(

ESLint: 8.43.0

TypeError: Cannot convert undefined or null to object
Occurred while linting <root>/packages/design-system/api/getTranscript.ts:3
Rule: "turbo/no-undeclared-env-vars"
    at Function.values (<anonymous>)
    at xe.test (<root>/node_modules/eslint-plugin-turbo/dist/index.js:84:5623)
    at a (<root>/node_modules/eslint-plugin-turbo/dist/index.js:84:6743)
    at MemberExpression (<root>/node_modules/eslint-plugin-turbo/dist/index.js:84:7630)
    at ruleErrorHandler (<root>/node_modules/eslint/lib/linter/linter.js:1050:28)
    at <root>/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (<root>/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (<root>/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (<root>/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
npm ERR! Lifecycle script `lint` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: @learning-destination/design-system@0.0.0 
npm ERR!   at location: <root>/packages/design-system 
ERROR: command finished with error: command (<root>/packages/design-system) npm run lint exited (1)
```

And it's pointing to an environment variable that _is_ declared in turbo.json.

### Reproduction Repo

_No response_

--- a/packages/eslint-plugin-turbo/lib/utils/calculate-inputs.ts
+++ b/packages/eslint-plugin-turbo/lib/utils/calculate-inputs.ts
@@ -436,7 +436,7 @@ export class Project {
       ),
     ];
 
-    if (workspaceName) {
+    if (workspaceName && workspaceName in this._test.workspaceTasks) {
       tests.push(
         ...Object.values(this._test.workspaceTasks[workspaceName]).map(
           (context) => environmentTestArray(context)

